steps/rabbit/steps.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbit
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Telefonica/golium"
+	"github.com/cucumber/godog"
+)
+
+// Steps to initialize common steps.
+type Steps struct {
+}
+
+// InitializeSteps initializes all the steps.
+func (cs Steps) InitializeSteps(ctx context.Context, scenCtx *godog.ScenarioContext) context.Context {
+	// Initialize the HTTP session in the context
+	ctx = InitializeContext(ctx)
+	session := GetSession(ctx)
+	// Initialize the steps
+	scenCtx.Step(`^the rabbit endpoint "([^"]*)"$`, func(uri string) error {
+		return session.ConfigureConnection(ctx, uri)
+	})
+
+	scenCtx.Step(`^I subscribe to the rabbit topic "([^"]*)"$`, func(topic string) error {
+		return session.SubscribeTopic(ctx, golium.ValueAsString(ctx, topic))
+	})
+	scenCtx.Step(`^I unsubscribe from the rabbit topic "([^"]*)"$`, func(topic string) error {
+		return session.UnsubscribeTopic(ctx, golium.ValueAsString(ctx, topic))
+	})
+	scenCtx.Step(`^I publish a message to the rabbit topic "([^"]*)" with the text$`, func(topic string, message *godog.DocString) error {
+		return session.PublishTextMessage(ctx, golium.ValueAsString(ctx, topic), golium.ValueAsString(ctx, message.Content))
+	})
+	scenCtx.Step(`^I publish a message to the rabbit topic "([^"]*)" with the JSON properties$`, func(topic string, t *godog.Table) error {
+		props, err := golium.ConvertTableToMap(ctx, t)
```

Comment:
It seems to be a problem when the table has array type values. They are processed as raw string values:
![image](https://user-images.githubusercontent.com/9568287/114716650-8b8bfe80-9d34-11eb-9cfa-350f058a2949.png)

![image](https://user-images.githubusercontent.com/9568287/114716781-aeb6ae00-9d34-11eb-8ae3-cb044e83e059.png)

Reply:
So far, no list has ever been sent within a table of the kind you mention. The table maps as the content is in the key-value format. Then, the value is casted to string. If you use the next step, you can't send lists:
`I publish a message to the rabbit topic "test-topic" with the JSON properties.`

---

I have implemented a step to send a JSON directly (`I publish a message to the rabbit topic "test-topic" with the JSON`) and the JSON is not sent in table format.

Example:

```json
When I publish a message to the rabbit topic "test-topic" with the JSON
  {
    "key1": "value1",   
    "listA": ["element1A", "element2A", "element3A"],
    "listB": ["element1B", "element2B", "element3B"],
    "nestedElement": {
         "nestA": "value1"
    }
  }
```

On the other hand, there are two ways to test the received message. The first is to check the full JSON with the step (`I wait up to "3" seconds for a rabbit message with the JSON`). The second is to check the JSON fields one by one with the step (`I wait up to "3" seconds for a rabbit message with the JSON properties`).

Take a look at it when you can and if you have any ideas, we can discuss them together.